### ISSUE TITLE
[apple.getPurchaseData] incorrect expirationDate after canceling subscription

### ISSUE THREAD

**Issue by jbgomez21** [voltrue2/in-app-purchase#137]
Relative to https://github.com/voltrue2/in-app-purchase/issues/124

This changed was done for lib/google.js but not in lib/apple.js, that is right?

```
function getSubscriptionExpireDate(data) {
	if (!data) {
		return 0;
	}
	if (data[REC_KEYS.CANCELLATION_DATE] && data[REC_KEYS.EXPIRES_DATE] !== data[REC_KEYS.CANCELLATION_DATE]) {
		return data[REC_KEYS.CANCELLATION_DATE];
	}
	if (data[REC_KEYS.EXPIRES_DATE_MS]) {
		return parseInt(data[REC_KEYS.EXPIRES_DATE_MS], 10);
	}
	if (data[REC_KEYS.EXPIRES_DATE]) {
		return data[REC_KEYS.EXPIRES_DATE];
	}
	if (data[REC_KEYS.EXPIRATION_DATE]) {
		return data[REC_KEYS.EXPIRATION_DATE];
	}
	if (data[REC_KEYS.EXPIRATION_INTENT]) {
		return parseInt(data[REC_KEYS.EXPIRATION_INTENT], 10);
	}
	return 0;
}
```

The next lines should be deleted?

```
	if (data[REC_KEYS.CANCELLATION_DATE] && data[REC_KEYS.EXPIRES_DATE] !== data[REC_KEYS.CANCELLATION_DATE]) {
		return data[REC_KEYS.CANCELLATION_DATE];
	}
```
I don't wanna change the expirationDate if purchase is cancelled.

Regards

**Comment by voltrue2** [comment]
Hello

Thank you for reaching out. I will look into this as soon as I get the chance.

Cheers!

**Comment by jbgomez21** [comment]
Hi @voltrue2, 

Is it too much trouble if I ask you to take a look at this issue?
It's too important for auto-renew subscription.
Users usually subscribe and cancel his membership for not renew the next time.
With this bug, after cancel the server return that his purchase have expired.

Thx!

Regards!

**Comment by voltrue2** [comment]
Hello

Thank you for reaching out.

It does sound confusing... Let me read this a little more carefully and I will try to get back to you on this ASAP.

Cheers!

**Comment by triplef** [comment]
I think this issue was fixed in #175 via this commit: https://github.com/voltrue2/in-app-purchase/commit/e0acff7ab5020e2f07739cf84399a78ed7938832

@jbgomez21 / @voltrue2 would be great if you can confirm.

**Comment by voltrue2** [comment]
Hello   
  

  
Yes. You’re right it has been fixed with the commit you mentioned.   
  
I’ll close the issue.
  

  
Cheers!
  

  
  

  
  
>
